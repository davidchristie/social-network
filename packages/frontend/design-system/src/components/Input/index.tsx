import FormControl from "@material-ui/core/FormControl";
import MaterialInput from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import React from "react";

interface Props {
  autoFocus?: boolean;
  className?: string;
  id: string;
  label: string;
  name: string;
  onBlur?: React.FocusEventHandler;
  onChange?: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  onClick?: React.MouseEventHandler;
  required?: boolean;
  type?: string;
  value?: number | string;
}

const Input: React.SFC<Props> = ({
  id,
  label,
  name,
  required,
  ...attributes
}) => {
  return (
    <FormControl fullWidth={true} margin="normal" required={required}>
      <InputLabel htmlFor={id}>{label}</InputLabel>
      <MaterialInput
        autoComplete={name}
        id={id}
        name={name}
        {...attributes}
      />
    </FormControl>
  );
};

export default Input;
