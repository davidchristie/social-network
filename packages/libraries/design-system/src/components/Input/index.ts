import React from "react";

import elementWithClass from "../../utilities/elementWithClass";
import "./index.css";

interface Props extends React.InputHTMLAttributes<HTMLInputElement> { }

const Input = elementWithClass<Props>("Input", "input");

export default Input;
