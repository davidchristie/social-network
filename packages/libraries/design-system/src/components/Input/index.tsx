import MaterialInput from "@material-ui/core/Input";
import React from "react";

interface Props {
  className?: string;
  onBlur?: (event: React.FocusEvent) => void;
  onChange?: (event: React.FormEvent) => void;
  onClick?: (event: React.MouseEvent) => void;
}

const Input: React.SFC<Props> = props => (
  <MaterialInput {...props} />
);

export default Input;
