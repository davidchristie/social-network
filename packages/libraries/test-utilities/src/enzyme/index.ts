export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
