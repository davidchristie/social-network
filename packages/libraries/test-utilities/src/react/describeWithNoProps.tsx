import React from "react";

import describeWithProps from "./describeWithProps";

export default function describeWithNoProps (
  Component: React.ComponentType
) {
  describeWithProps("with no props", Component, {});
}
