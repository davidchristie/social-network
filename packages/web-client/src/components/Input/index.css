.Input {
  font-size: 1.4em;
  width: 100%;
}
