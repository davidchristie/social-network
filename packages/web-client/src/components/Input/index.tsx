import React from "react";

import "./index.css";

interface Props extends React.InputHTMLAttributes<HTMLInputElement> { }

export default class Input extends React.Component<Props> {
  public render () {
    const { className, ...attributes } = this.props;
    return (
      <input
        className={`Input${className ? " " + className : ""}`}
        {...attributes}
      />
    );
  }
}
