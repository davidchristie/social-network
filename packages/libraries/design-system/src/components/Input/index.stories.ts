import { action } from "@storybook/addon-actions";

import Input from ".";
import addComponentStory from "../../utilities/storybook/addComponentStory";

addComponentStory(Input, {
  label: "Input",
  name: "input",
  onBlur: action("onBlur"),
  onChange: action("onChange"),
  onClick: action("onClick"),
});
