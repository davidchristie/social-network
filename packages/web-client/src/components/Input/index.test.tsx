import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import describeWithNoProps from "../../testings/describeWithNoProps";
import Input from "./index";

describe("Input component", () => {
  let wrapper: ShallowWrapper;

  describeWithNoProps(Input);

  describe("with custom class", () => {
    const CUSTOM_CLASS = "custom_class";

    beforeEach(() => {
      wrapper = shallow(
        <Input className={CUSTOM_CLASS} />,
      );
    });

    it("has custom class", () => {
      expect(wrapper.hasClass(CUSTOM_CLASS)).toBe(true);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
