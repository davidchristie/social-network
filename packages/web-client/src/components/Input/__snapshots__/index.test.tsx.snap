// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component with custom class matches snapshot 1`] = `
<input
  className="Input custom_class"
/>
`;

exports[`Input component with no props matches snapshot 1`] = `
<input
  className="Input"
/>
`;
